vpr/src/base/read_blif.cpp

```diff
@@ -666,6 +673,70 @@ vtr::LogicValue to_vtr_logic_value(blifparse::LogicValue val) {
     return new_val;
 }
 
+bool is_string_param(const std::string& param) {
+    /* Empty param is considered a string */
+    if (param.empty()) {
+        return true;
+    }
+
+    /* There have to be at least 2 characters (the quotes) */
+    if (param.length() < 2) {
+        return false;
+    }
+
+    /* The first and the last characters must be quotes */
+    size_t len = param.length();
+    if (param[0] != '"' || param[len - 1] != '"') {
+        return false;
+    }
+
+    /* There mustn't be any other quotes except for escaped ones */
+    for (size_t i = 1; i < (len - 1); ++i) {
+        if (param[i] == '"' && param[i - 1] != '\\') {
+            return false;
+        }
+    }
+
+    /* This is a string param */
+    return true;
+}
+
+bool is_binary_param(const std::string& param) {
+    /* Must be non-empty */
+    if (param.empty()) {
+        return false;
+    }
+
+    /* The string must contain only '0' and '1' */
+    for (size_t i = 0; i < param.length(); ++i) {
+        if (param[i] != '0' && param[i] != '1') {
+            return false;
+        }
+    }
+
+    /* This is a binary word param */
+    return true;
+}
+
+bool is_real_param(const std::string& param) {
+    const std::string chars = "012345678.";
```

Comment:
Missing 9 --> need to add it to "chars".
Suggest using a more descriptive name than chars. real_chars would be better.

Reply:
Right... stupid mistake. Fixed.

Reply:
I gave it more thought and made the function use a regular expression.